ID: 47587
Title: global sort import 1T data fail for "[0]runtime error: makeslice: len out of range" 
Description:
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. set global tidb_cloud_storage_uri="s3://xxx"
2. import into globalsort.item_core_2 FROM "s3://xxx&force-path-style=false&region=Beijing&provider=ks" with thread=16,detached;

### 2. What did you expect to see? (Required)
import successfully

### 3. What did you see instead (Required)
```
+--------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------+----------+-----------+--------+------------------+---------------+-----------------------------------------------+----------------------------+----------------------------+----------------------------+------------+
| Job_ID | Data_Source                                                                                                                                                                                                     | Target_Table               | Table_ID | Phase     | Status | Source_File_Size | Imported_Rows | Result_Message                                | Create_Time                | Start_Time                 | End_Time                   | Created_By |
+--------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------+----------+-----------+--------+------------------+---------------+-----------------------------------------------+----------------------------+----------------------------+----------------------------+------------+
| 150001 | s3://qe-testing/xxx/10T/data3/test.item_core.*.csv?access-key=xxxxxx&endpoint=xxx&force-path-style=false&provider=ks&region=Beijing&secret-access-key=xxxxxx | `globalsort`.`item_core_2` |      163 | importing | failed | 1.044TB          |          NULL | [0]runtime error: makeslice: len out of range | 2023-10-12 10:54:49.935207 | 2023-10-12 10:54:51.070273 | 2023-10-12 13:49:36.077075 | root@%     |
+--------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------+----------+-----------+--------+------------------+---------------+-----------------------------------------------+----------------------------+----------------------------+----------------------------+------------+
1 row in set (0.01 sec)

MySQL [(none)]> 
```

### 4. What is your TiDB version? (Required)
3eadac40199a11f74addb3fe056fdc0ea6ca3b6d

<!-- Paste the output of SELECT tidb_version() -->

